### DISCUSSION TITLE
Out_of_Date targets when downloaded from Cluster

### DISCUSSION THREAD

**Discussion by Skourtis** [ropensci/targets#370] [Help] [answered]
Hi,

I'm running my _targets.R on an SGE cluster and then copy everything locally to work locally on it. But everything is out of date. I also tried going through git by pulling the targets/meta and targets/objects but still everything is out of date on my local machine. Is the an issue only for me or a known issue? Locally I'm working on Windows 10. Could this be a cluster-linux, local-windows issue?

How do I use targets if I want to have collaborators on git who need to know that the targets they see are up to date?

Thank you?
Savvas

**Comment by wlandau** [comment, marked as answer]
It is hard to say. Does the pipeline have any [external input files](https://books.ropensci.org/targets/files.html#external-input-files) that are system-dependent or missing when you just copy over `_targets/`? Does `tar_visnetwork()` point to anything specific upstream that might be invalidating the rest of the targets? It would help if you would share a small reproducible example.

**Reply by Skourtis** [reply]
Hi, it is as you predicted.  After some digging I realised that the path structure are different in my local drive and cluster. I'm using here::here to create relative paths in tar_target which are identical for both, but when this is saved as a variable using the format = "file", it creates and saves the absolute path. So when I pull from git, it seems that I have changed the input files (which I haven't). I think this is the problem but not sure how to confirm or solve it. Let's say I am using this to collaborate through git, I cannot expect them to have the same absolute path as me (which is why I was using here::here).

**Reply by wlandau** [reply]
`here::here()` creates an absolute path. For `targets`, it is better to use a relative path. I recommend something like `fs::path_rel(here::here("..."))` instead.

**Reply by Skourtis** [reply]
This has solved the issue! thank you!
